### ISSUE TITLE
<Space> doesn't complete variable name, when Variable explictly selected from IntelliSense list

### ISSUE THREAD

**Issue by ghost** [microsoft/poshtools#551]
OS: Win Svr 2012 R2
VS: Dev12 Ent + Update5
Powershell v4.0 + latest PoshTool

Repro Steps:
1. Create a PowerShell Script Project
2. Set a variable like $testVariable
3. Press $ to enable intelliSense and choose the variable that you just setting
4. Press `<Space>`

Expect: auto-complete the variable name
Actual: Just input a blank and not complete variable name

**Comment by ASayre** [comment]
Thanks for reporting this!  However, this is by design, as we don't necessarily know if this is the first usage of the variable or not. 

It would be difficult to type 

```
$apple = "potato"

$app = 42
```

If we auto completed after hitting space on $app.

Please reopen if you disagree!

**Comment by ghost** [comment]
Just like I said in repro step 3:  Press $ to enable intelliSense and "choose the variable that you just setting." We have choose the variable that we want to use in intelliSense and just want to input it quickly. Like we could press "Enter" to input complete variable quickly after you choose it in intelliSense already but just will input a blank after press <Space> at here.
![alreadychoose](https://cloud.githubusercontent.com/assets/12426589/7677738/3d8e315c-fd81-11e4-9060-3429755e351b.JPG)

**Comment by ASayre** [comment]
Reopening as @ArcherSui is correct; after we explicitly select an item, hitting space should complete it.

**Comment by HuanhuanSunMSFT** [comment]
No, we shouldn't complete the variable with SPACE even you explicityly select an item. Take $apple and then $app as an example, the $app is actually selected when you're typing. Besides, this way we are in parity with ISE. @AndreSayreMSFT @ArcherSui

**Comment by gwojan** [comment]
I would like to argue that you _should_ allow complete with space. Personally, I really have issues with using the ISE as the standard. PoshTools is a Visual Studio extension so it should behave like native Visual Studio not the ISE. Just my humble opinion. :smirk:

**Comment by pshamus** [comment]
I agree with @gwojan, PoshTools should by default behave like native Visual Studio, not the ISE.  That is the whole reason why I want to (and currently) use VS as my development environment: to have consistent behavior no matter the language.  There are many aspects of the ISE that I greatly dislike, and I am hoping PowerShell can be a true first-class citizen within VS.  I already submitted a request (#474) to give people the option to turn off ISE-compatibility.

Just my $0.02.

**Comment by ASayre** [comment]
If an item is selected in the completion list, only then should 'space' auto complete it.
